### PR TITLE
fix: differentiate auth errors for missing token vs no access

### PR DESCRIPTION
**Pull request by jomifepe**
## Summary
- Replace the blanket "Not logged in" message for all 401/403 errors with context-aware auth messages based on token presence and error type
- Add `formatAuthErrorMessage()` in the shared request layer so the global handler can distinguish not logged in, invalid/expired token, and repository access denied
- Add unit and e2e tests covering the new message paths

## Test plan
- [x] `npm run unit -- test/auth-errors.test.ts test/whoami.test.ts`
- [x] `npm run test`

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: src/lib/request.ts [hunk 1]
```diff
@@ -113,3 +113,19 @@ export class ForbiddenRequestError extends RequestError {
...
+	if (!options.hasToken) {
+		return "Not logged in. Run `prismic login` first.";
+	}
+	if (error instanceof UnauthorizedRequestError || options.envToken) {
+		if (options.envToken) {
+			return "PRISMIC_TOKEN is invalid or expired. Unset it to log in with a browser, or replace it with a valid token.";
+		}
+		return "Your session is invalid or expired. Run `prismic login` to sign in again.";
+	}
+	return "You do not have access to this repository. Check the repository name or log in with an account that has access.";
```
Comment (levimykel) [inline src/lib/request.ts]: ❓ When PRISMIC_TOKEN is set, a 403 returns “PRISMIC_TOKEN is invalid or expired” — but 403 usually means the token is valid and just lacks access to this repo, so the L130 “no access” message never fires for env-token users. Does our API return 401 for bad/expired tokens and 403 only for no-access? If so, 403 should fall through to the access-denied message even when an env token is present. If the backend returns 403 for both, current behavior is fine — just want to confirm.

Comment (levimykel) [reply src/lib/request.ts]: 💡 Depending on what you want here, this could be simplified to something like this, yeah?

```
if (!options.hasToken) return "Not logged in…";
if (options.envToken) return "PRISMIC_TOKEN is invalid or expired…";
if (error instanceof UnauthorizedRequestError) return "Your session is invalid or expired…";
return "You do not have access to this repository…";
```

**Review by levimykel — APPROVED** [review]
Approved with a small question/comment.

File: src/lib/request.ts (hunk 1 above)
Comment (jomifepe) [reply src/lib/request.ts]: Yeah, I agree that the `options.envToken` would be better only checked inside the condition. Fixed that ✅

Regarding the messages, even though I like the shorter messages, I think it's good to give the user a suggestion, to provide a "way out" of the error state.

Comment (levimykel) [reply src/lib/request.ts]: Okay, sounds good 👍

File: src/lib/request.ts [hunk 2]
```diff
@@ -113,3 +114,21 @@ export class ForbiddenRequestError extends RequestError {
 export class UnauthorizedRequestError extends RequestError {
 	name = "UnauthorizedRequestError";
 }
+
+export function formatAuthErrorMessage(
```
Comment (angeloashmore) [inline src/lib/request.ts]: `src/lib/request.ts` is (mostly) domain-agnostic. All of `lib` is supposed to be generic without Prismic knowledge to keep the functions flexible and reusable. (Sorry, I know this isn't documented anywhere. Maybe we should add some docs.)

Could we move `formatAuthErrorMessage` somewhere else? Because it's only used once in the code base, and we handle auth errors globally, maybe we can inline this code in `src/index.ts`. WDYT?

File: test/auth-errors.test.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+afterEach(() => {
+	env.PRISMIC_TOKEN = undefined;
+});
+
+function mockResponse(status: number): Response {
+	return new Response(null, { status, statusText: status === 401 ? "Unauthorized" : "Forbidden" });
+}
+
+describe("formatAuthErrorMessage", () => {
```
Comment (angeloashmore) [inline test/auth-errors.test.ts]: In general, we've favored integration/e2e tests over unit tests. Is there a way to test the changes through a command rather than checking the implementation?

We can make an exception if testing these edge cases is difficult.

**Review by angeloashmore — APPROVED** [review]
Looks good!
